### PR TITLE
client: support all admin apis

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

**Review by bailantaotao — APPROVED** [review]

File: client/admin.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	"context"
+
+	"github.com/ethereum/go-ethereum/p2p"
+)
+
+// AddPeer connects to the given nodeURL.
+func (c *client) AddPeer(ctx context.Context, nodeURL string) (bool, error) {
+	var r bool
+	// TODO: Result needs to be verified
+	// The response data type are bytes, but we cannot parse...
```
Comment (yutelin) [inline client/admin.go]: Is it still a TODO item?
It seems we obtain the response data and return it as a `bool`

File: client/admin.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	}
+	return r, nil
+}
+
+// RemovePeer disconnects from a a remote node if the connection exists
+func (c *client) RemovePeer(ctx context.Context, nodeURL string) (bool, error) {
+	var r bool
+	// TODO: Result needs to be verified
+	// The response data type are bytes, but we cannot parse...
+	err := c.rpc.CallContext(ctx, &r, "admin_removePeer", nodeURL)
```
Comment (yutelin) [inline client/admin.go]: ditto

File: client/admin.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		return false, err
+	}
+	return r, nil
+}
+
+// Peers returns the number of connected peers.
+func (c *client) Peers(ctx context.Context) ([]*p2p.PeerInfo, error) {
+	var r []*p2p.PeerInfo
+	// The response data type are bytes, but we cannot parse...
+	err := c.rpc.CallContext(ctx, &r, "admin_peers")
```
Comment (yutelin) [inline client/admin.go]: ditto

File: client/admin.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	// TODO: Result needs to be verified
+	// The response data type are bytes, but we cannot parse...
+	err := c.rpc.CallContext(ctx, &r, "admin_addPeer", nodeURL)
+	if err != nil {
+		return false, err
+	}
+	return r, nil
+}
+
+// RemovePeer disconnects from a a remote node if the connection exists
```
Comment (yutelin) [inline client/admin.go]: remove a `a`

File: client/admin.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	// TODO: Result needs to be verified
+	// The response data type are bytes, but we cannot parse...
+	err := c.rpc.CallContext(ctx, &r, "admin_removePeer", nodeURL)
+	if err != nil {
+		return false, err
+	}
+	return r, nil
+}
+
+// Peers returns the number of connected peers.
```
Comment (yutelin) [inline client/admin.go]: return the number of peers or the peers

**Review by yutelin — APPROVED** [review]

File: client/admin.go (hunk 1 above)
Comment (markya0616) [reply client/admin.go]: We should remove it

File: client/admin.go (hunk 2 above)
Comment (markya0616) [reply client/admin.go]: We should remove todo.

File: client/admin.go (hunk 3 above)
Comment (markya0616) [reply client/admin.go]: We should remove todo.

File: client/admin.go (hunk 4 above)
Comment (markya0616) [reply client/admin.go]: OK.

File: client/admin.go (hunk 5 above)
Comment (markya0616) [reply client/admin.go]: OK.

**Review by tailingchen — APPROVED** [review]

**Pull request closed without merging**